ops/deployment/docker-compose.production-envs.yml

```diff
@@ -25,6 +25,7 @@ services:
       - ${DATA_SAVE_PATH}/fuw/repo:/var/www/filedrop
       # public download path (TODO: temporary, to revisit in storage work stream)
       - /tmp:/var/www/datasetfiles
+      - templates:/var/www/files/templates
```

Comment:
Hi @pli888 and @rija ,

Need to specify this volume, otherwise will get this error on staging like:
```
Error 404
Unable to resolve the request "files/templates/GigaDBUploadForm-forWebsite-v22Dec2021.xlsx".
```

Reply:
Hi @kencho51, 

For production environments the protocol is to copy static files inside container image by using COPY in ``Production-Web-Dockerfile``.

You can see that being done already for the files in ``files/examples/``.

So you follow the same model for files in ``files/templates``.

The idea is that a container image is like a CD-ROM, a golden disc image that  contains a readonly snapshot of all our files, make it easier to rollback in case of problems, know exactly what set of file a  particular deployment consist of and for auditabilty.

So we copy all the files of the codebase into those immutable images: php code goes into ``Production-dockerfile``, and static files (because they are served by nginx only, not php-fpm) go into ``Production-Web-Dockerfile``

We only use volumes for when we know the deployed application needs to write something to the filesystem (like the Let's Encrypt certificates or some dynamic assets created by the Yii framework)

Reply:
Hi @rija,

The `- templates:/var/www/files/templates` has been removed. After rebuilding the containers in staging server through gitlab pipeline, the template files download works as expected.

Reply:
@kencho51 you sure about that? 
how could it work on staging if you don't copy them inside the production container image?